log/log.go

```diff
@@ -19,6 +23,33 @@ func Error(msg string, err error) {
 	Logger.Error(msg, err)
 }
 
+func Errorf(err error, format string, v ...interface{}) {
+	Logger.Errorf(format, err, v...)
+}
+
+// ErrorWithSpan gets the span from the provided context.
+// Used for Datadog.
+func ErrorWithSpan(ctx context.Context, msg string, err error) {
+	span, ok := tracer.SpanFromContext(ctx)
+	if !ok {
+		Error(msg, err)
+		return
+	}
+
+	Errorf(err, msg+", span: %v", span)
+}
```

Comment:
This is a proof of concept for datadog's integration. Another way would be to just call Errorf() from any service and make sure we add the `span: %v`.

Of course, adding datadog things here simplifies things for the other services at the expense of complicating things in obs (just look at the go.mod now). At the end of the day it's a choice.